### PR TITLE
Add "Transaction Id" and "Client Version" in Diagnostic Source traces

### PR DESCRIPTION
**Pull request by cheenamalhotra**
Addressing request made by @davidmikh here: https://github.com/dotnet/SqlClient/issues/93#issuecomment-590941699

Hi @davidmikh 

As per our offline discussions, I've made this PR to include "Transaction Id" to `SqlCommand` and `SqlTransaction` events and "Client Version" in `SqlConnection` events.

Regarding other 2 requests:
* Include _Connection Id_ in `SqlTransaction` events
  -- `Connection.ClientConnectionId` contains this info.
* Include _Server Version_ in `SqlConnection` open events
  -- `Connection.ServerVersion` contains this info.

Let me know if you need additional info from here.

P.S. Diagnostic Tests to be enabled in a separate PR (WIP @JRahnama PR #517)

### CODE REVIEW

**Comment by cheenamalhotra** [comment]
Uploading NuGet Package for validation:
[Microsoft.Data.SqlClient.2.0.0-dev-pr515.zip](https://github.com/dotnet/SqlClient/files/4457238/Microsoft.Data.SqlClient.2.0.0-dev-pr515.zip)

Usage: `<PackageReference Include="Microsoft.Data.SqlClient" Version="2.0.0-dev-pr515" />`

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlClientDiagnosticListenerExtensions.cs [hunk 1]
```diff
@@ -52,6 +52,7 @@ public static Guid WriteCommandBefore(this DiagnosticListener @this, SqlCommand
                         Operation = operation,
                         ConnectionId = sqlCommand.Connection?.ClientConnectionId,
                         Command = sqlCommand,
+                        TransactionId = transaction?.InternalTransaction?.TransactionId,
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlClientDiagnosticListenerExtensions.cs:55]: These might be able to be simplified to just:
```suggestion
                        transaction?.InternalTransaction?.TransactionId,
```

Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlClientDiagnosticListenerExtensions.cs]: Done!

**Comment by davidmikh** [comment]
👍 Looks good to me, thanks for making these changes!

**Pull request merged**